packages/markdown-docx/src/ToOOXMLVisitor/index.js

```diff
@@ -51,6 +52,23 @@ class ToOOXMLVisitor {
         this.tags = [];
         // Relationship tags for links in a document
         this.relationships = [];
+        // Contains the tags for optional or conditional nodes.
+        // Optional/conditional themselves can act contain multiple inline nodes.
+        // However, they are not block elements and are present inside
+        // paragraph or clause nodes so an extra array is needed to store their info.
+        this.conditionalOrOptionalTags = [];
+
+        // Whether the nodes traversed in conditional or optional are of false condition
+        // in optional this means if hasSome is true then whenNone are these nodes
+        // if hasSome is false then whenSome nodes are the corresponding required nodes
+        // this is done to add vanish property
+        this.isTraversingFalseCondition = false;
+
+        // Whether the nodes traversed are present in else condition
+        // done to add specific font property
+        // font property+ vanish = whenNone, else nodes when converting from
+        // ooxml->ciceromark. simple vanish means whenNone, if nodes
+        this.isTraversingElseCondition = false;
```

Comment:
It's quite dangerous to have class variables like these and then use a method to flip their values. I would recommend you to make this function local to the block where you check if the transformed node is `Optional`.